specs/SharedBuffer.md

```diff
@@ -0,0 +1,310 @@
+Shared Buffer Between Native Application Code and Script
+===
+
+# Background
+For some advanced scenarios, there is a need to exchange large data between WebView application process and trusted web pages that are considered as part of the app. Some examples:
+- Web page generates large data, passes to native side to be further process or fed to other parts of the app or OS. Example: https://github.com/MicrosoftEdge/WebView2Feedback/issues/89.
+-	Native side generates large data for the web side to consume. The data might or might not come directly from files. Example: https://github.com/MicrosoftEdge/WebView2Feedback/issues/1005.
+To support these scenarios, we are adding an Edge WebView2 API to support shared buffer between app and renderer process, based on shared memory from the OS.
+
+The application code can use the APIs to create a shared buffer object, and share to scripts as [ArrayBuffer](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/ArrayBuffer) object.
+Then both the native application code and the script will be able to access the same memory.
+
+# Conceptual pages (How To)
+
+Besides using the memory address directly, the shared buffer object can be accessed from native side via an IStream* object that you can get from the shared object.
+
+When the application code calls `PostSharedBufferToScript`, the script side will receive a `SharedBufferReceived` event containing the buffer as an `ArrayBuffer` object.
+After receiving the shared buffer object, it can access it the same way as any other ArrayBuffer object, including transering to a web worker to process the data on
+the worker thread.
+
+As shared buffer normally represent a large memory, instead of waiting for garbage collection to release the memory along with the owning objects, the application
+should try to release the buffer as soon as it doesn't need access to it. This can be done by calling Close() method on the shared buffer object, or
+`chrome.webview.releaseSharedBuffer` from script.
+
+As the memory could contain sensitive information and corrupted memory could crash the application, the application should only share buffer with trusted sites.
+
+The application can use other messaging channel like `PostWebMessageAsJson` and `chrome.webview.postMessage` to inform the other side the desire to have a shared
+buffer and the status of the buffer (data is produced or consumed).
+
+# Examples
+
+The example below illustrates how to send data from application to script for one time read only consumption.
+
+The script code will look like this:
+```
+        window.onload = function () {
```

Comment:
Why wait until onload - does that create a race?

In the native code you don't show when the PostSharedBufferToScript call is done regarding the status of the document loading in the WebView2. If you set the JS event handler too late isn't there a chance you've set the event handler after the call to PostSharedBufferToScript?